lib/core/src/Cardano/Wallet/Primitive/AddressDerivation/Random.hs

```diff
@@ -135,3 +152,36 @@ hashSeed = serialize . blake2b256 . serialize
 
 blake2b256 :: ScrubbedBytes -> ScrubbedBytes
 blake2b256 = BA.convert . hash @ScrubbedBytes @Blake2b_256
+
+-- | Compute passphrase as hash of the root public key.
+deriveHDPassphrase :: Key 'RootK XPub -> Passphrase "random"
+deriveHDPassphrase (Key pk) = Passphrase $
+    PBKDF2.generate
+        (PBKDF2.prfHMAC SHA512)
+        (PBKDF2.Parameters
+             500 -- Parameters for the hashing function. 500 iter of PBDKF2 with HMAC-SHA256
+             passLen)
+        (unXPub pk)
+        ("address-hashing"::ByteString)
+  where
+    -- Password length in bytes
+    passLen = 32
+
+eskToHdPassphrase :: Key 'RootK XPrv -> Passphrase "random"
+eskToHdPassphrase = deriveHDPassphrase . publicKey
+
+decryptDerPath
+    :: Passphrase "random"
+       -- ^ Symmetric key / passphrase, 32-byte long
+    -> Address
+    -> CryptoFailable ByteString
```

Comment:
It is questionable that we want to have a `CryptoFailable` here as a top-level value. Depends if we consider this to be an internal function or not, but I'd rather not spread this across many modules..